run_test.go

```diff
@@ -20,8 +20,22 @@ func testRun(t *testing.T, context spec.G, it spec.S) {
 		tmpDir      string
 		cnbDir      string
 		exitHandler *fakes.ExitHandler
+
+		buildCalled  bool
+		detectCalled bool
 	)
 
+	var build packit.BuildFunc
```

Comment:
This line is actually needed because of the introduction of `_BuildOrGenerate`. So strictly speaking, this would be a breaking change, or?
@ryanmoran WDYT?

Reply:
You are right. I didn't run into this in my testing because we tend to do something like this:

```go
func Build() packit.BuildFunc {
	return func(context packit.BuildContext) (packit.BuildResult, error) { ... }
}
```

I guess I am less caught up in trying to make this API read nicely. I'd be fine with 2 Run funcs.